src/reverb/classes/models/ReverbMapping.php

```diff
@@ -188,10 +195,69 @@ private function insertMapping($psCategoryId, $reverbCode)
         );
         if ($exec) {
             $return = Db::getInstance()->Insert_ID();
+            if ($return) {
+                $return = $this->subCategoriesManagement($psCategoryId, $reverbCode);
+            }
         }
         $this->module->logs->infoLogs(
             'Insert mapping for category ' . $psCategoryId . ' => ' . $reverbCode . ' : result = ' . var_export($return, true)
         );
         return $return;
     }
+
+    /**
+     * Delete and insert the new Reverb category
+     *
+     * @param $psCategoryId
+     * $return bool
+     */
+
+    private function subCategoriesManagement($psCategoryId, $reverbCode)
+    {
+        $return = false;
+        $delete_categories = array();
+        $insert_categories = array();
+        // load the category object
+        $cat = new Category($psCategoryId);
+        // get all category children
+        $all_cat = $cat->getAllChildren();
```

Comment:
I want to make sure we're doing the right thing here. We will definitely have more than one tier of categories. Take cymbals: https://reverb.com/marketplace/drums-and-percussion/cymbals . You cannot set cymbals as a category on your listing, but you can set one of its children. Not sure which API call you are making here, but I do want to make sure we're doing the right thing.

Reply:
Hi @kylecrum ,

This call recovers all children to the selected category PrestaShop. It's not a call to Reverb API.

To optimize the screen in the Reverb module configuration for seller when they must to map the Reverb categories with Prestashop categories, we show only main categories (first level) because a lot of seller have more 3000 categories to map... When we map a Prestashop category with a reverb code, we get all children to the Prestashop category and map with the same reverb code. 
So, we will have to see with Aabesh to optimize again this screen to configure a reverb code specific to children. I made a mockup for that and a trello ticket: https://trello.com/c/7Dl8nleG/63-optimize-category-mapping-for-children

Reply:
3000 categories :(